crypto/BRCryptoWalletManager.c

```diff
@@ -669,7 +677,20 @@ cryptoWalletManagerConnect (BRCryptoWalletManager cwm,
             ewmConnect (cwm->u.eth);
             break;
         case BLOCK_CHAIN_TYPE_GEN:
-            genManagerConnect(cwm->u.gen);
+            if (!genManagerIsConnected (cwm->u.gen)) {
```

Comment:
I guess this ship has sailed but this approach (setting state directly in calls rather than in single-threaded event callback) seems very racy?

Reply:
Addressed, I think, in this case.  Certainly inconsistencies elsewhere for GEN (if not others).